client/secureclient/src/main/java/com/linkedin/openhouse/client/ssl/WebClientFactory.java

```diff
@@ -154,14 +154,16 @@ private WebClient createSecureWebClient(String baseUrl, String truststoreLocatio
               .secure(t -> t.sslContext(createSslContext(truststore)));
     }
 
-    ClientHttpConnector connector = new ReactorClientHttpConnector(client);
+    return client;
+  }
+
+  private WebClient.Builder getWebClientBuilder(String baseUrl) {
     WebClient.Builder webClientBuilder = createWebClientBuilder();
     setSessionIdInWebClientHeader(webClientBuilder);
+    setClientNameInWebClientHeader(webClientBuilder);
```

Comment:
this is new: will take clientName method and pass to defaultHeader of WebClient.

WebClient will automagically pass defaultHeaders to all http requests  (or so I read on documentation, unfortunately i could not find a way to test this well)